Update LoginForm with 'cancel' option for 2FA

micschk opened the pull request:
FIX: #12

micschk committed:
Update LoginForm with 'cancel' option for 2FA

FIX: #12

camfindlay commented on code/form/LoginForm.php:
```diff
@@ -84,8 +94,9 @@ public function Actions()
         $fields = $this->Fields();
 
-        // Remove the lost-password action from the TOTP token form
+        // Remove the lost-password action from the TOTP token form and insert a cancel button
```
Seems like a good idea... common sense that once you've hit TOTP you've probably already remembered your password 👍

camfindlay reviewed (approved):
@oddnoc any issues with this implementation?

micschk committed:
Generate initial token if configured to not regenerate on Has2FA change.

Not too sure about the point where this should happen; updateCMSFields may not get called based upon implementation. populateDefaults() only gets called for new records (and would need a write action as well for the TOTP token to persist). requireDefaultRecords() would get called upon build, but not for newly created members. Hence the decision to implement on updateCMSFields.

camfindlay commented:
Will merge and tag as 2.1.0 if @oddnoc in agreement

camfindlay commented on code/extension/Member.php:
```diff
@@ -26,6 +26,8 @@ class Member extends \DataExtension
 
     private static $admins_can_disable = false;
 
+    private static $regenerate_on_activation = true;
```
These changes from #13 have snuck in here - can we make sure 1 PR = 1 feature, helps with review and faster merges ;)

Add the cancel button only on it's own branch and alter the PR branches here on GH trying to avoid a close and new PR if possible 👍

micschk replied:
Oops, sorry about that...

camfindlay replied:
No worries... lucky PRs are pretty malleable these days 😉

micschk committed:
Removing accidental regenerate_on_activation propety

micschk commented:
Things got a bit mixed up, replacement PR: #18
